bluetooth/src/androidLibMain/kotlin/BluetoothMonitor.kt

```
@@ -49,16 +50,17 @@ actual class BluetoothMonitor internal constructor(
         get() = bluetoothAdapter?.isEnabled == true
 
     override fun startMonitoring() {
-        if (bluetoothAdapter == null) return
+        if (bluetoothAdapter == null) warn("BluetoothMonitor") { "bluetoothAdapter is null" }
```

Does it return from function?

---

Also once it's subclass of ServiceMonitor you can call super here and in stopMonitoring as well

---

No it does not return. I think it should also print something because otherwise you don't understand why is not starting thee monitoring

---

`bluetoothAdapter` should not be optional, the class should throw an exception when being created.

---

530eca6